### PR TITLE
Feat/set subnet takes

### PR DESCRIPTION
**Pull request by distributedstatemachine**
## Description

This PR add function to set individual / multiple subnet takes. 

TODO:

- [x] Ensure they dont exceed the maximum
- [x] Ensure they can't violate the take change time limit

## Related Issue(s)

- Closes #[issue number]

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

## Breaking Change

If this PR introduces a breaking change, please provide a detailed description of the impact and the migration path for existing applications.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run `cargo fmt` and `cargo clippy` to ensure my code is formatted and linted correctly
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pallets/subtensor/src/staking.rs [hunk 1]
```diff
@@ -850,6 +850,98 @@ impl<T: Config> Pallet<T> {
...
+            take
+        );
+        // Check if the subnet exists before setting the take.
+        ensure!(
+            Self::if_subnet_exist(netuid),
+            Error::<T>::NetworkDoesNotExist
+        );
+
+ 
+                // --- 4. Ensure take is within the 0 ..= InitialDefaultTake (18%) range
```
Comment (gztensor) [inline pallets/subtensor/src/staking.rs]: The formatting is off

File: pallets/subtensor/src/staking.rs [hunk 2]
```diff
@@ -850,6 +850,98 @@ impl<T: Config> Pallet<T> {
...
+    ///
+    /// # Arguments
+    /// * `hotkey` - The account ID of the hotkey.
+    /// * `netuid` - The unique identifier for the network.
+    /// * `take` - The take rate to be set for the subnet.
+    ///
+    /// # Errors
+    /// Returns `Error::<T>::NetworkDoesNotExist` if the subnet does not exist.
+    /// Returns `Error::<T>::TakeExceedsDefault` if the take exceeds the initial default take.
+    pub fn do_set_delegate_take(origin: T::RuntimeOrigin, hotkey: &T::AccountId, netuid: u16, take: u16) -> dispatch::DispatchResult {
```
Comment (gztensor) [inline pallets/subtensor/src/staking.rs]: The method doesn't seem to be called in extrinsics. Do we really need it?

**Review by gztensor — CHANGES_REQUESTED** [review]

File: pallets/subtensor/src/staking.rs (hunk 1 above)
Comment (distributedstatemachine) [reply pallets/subtensor/src/staking.rs]: removed

File: pallets/subtensor/src/staking.rs (hunk 2 above)
Comment (distributedstatemachine) [reply pallets/subtensor/src/staking.rs]: removed

**Review by gztensor — APPROVED** [review]

**Pull request merged**